src/common/cockpitwebresponse.c

```diff
@@ -1572,7 +1572,7 @@ cockpit_web_response_negotiation (const gchar *path,
 
   while (!bytes)
     {
-      if (locale && shorter)
+      if (locale && shorter && g_strcmp0 (locale, shorter) != 0)
```

Comment:
as discussed on irc, this needs a comment (and maybe an updated commit message)

@mvollmer's suggestion: `only consider the shorter version of a locale when it is different from the full string`

I'm also not entirely sure about the logic of what is checked first.